### PR TITLE
Leave application notification

### PR DESCRIPTION
**Pull request by mukhil-geeky**

### CODE REVIEW

File: src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs [hunk 1]
```diff
@@ -164,6 +174,76 @@ async Task<bool> AddLeave()
                     AddLeaveRequestDayCommand addLeaveRequestDayCommand = new AddLeaveRequestDayCommand();
                     addLeaveRequestDayCommand.LeaveRequestId = leaveRequest.Id;
                     addLeaveRequestDayCommand.Date = dates;
+                    List<string> receivermanagerEmails = await _emailFinder.FindManagerEmailsAsync(employee.ManagerId, cancellationToken);
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs:177]: Done

File: src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs [hunk 2]
```diff
@@ -164,6 +174,76 @@ async Task<bool> AddLeave()
...
+                    List<string> SuperiorsMails = receivermanagerEmails.Skip(1).ToList();
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs:178]: done

File: src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs [hunk 3]
```diff
@@ -164,6 +174,76 @@ async Task<bool> AddLeave()
...
+                                    </html>";
+
+                      _email.SendMail(_senderEmail, _senderName, immediateManagerMail, message, subject, SuperiorsMails.ToArray());
+                    List<int> managerIds = await FindManagerIdsAsync(employee.ManagerId, cancellationToken);
+                    string notificationMessage = $" Leave {leaveRequest.FromDate} to {leaveRequest.ToDate} is Applied by {admin.FirstName} {admin.LastName} [Admin Code :{admin.EmployeeCode}] for {employee.FirstName} {employee.LastName} [Employee Code : {employee.EmployeeCode}]";
+                    foreach (int managerId in managerIds)
+                    {
+                        NotificationCommand command = new NotificationCommand
+                        {
+                            EmployeeId = managerId,
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs:242]: zameel ikka idhu, employee team undakiya command handler aanu. So update aakila

File: src/ui/src/app/dashboard/dashboard-body/main/leave-management/update-leave-settings/update-leave-settings.component.ts [hunk 4]
```diff
@@ -33,6 +33,7 @@ ngOnInit(){
   .pipe(takeUntilDestroyed(this.destroyRef)).subscribe({
     next:(response : ILeaveUpdate) =>{
       this.leaveSettings = response;
+      console.log(this.leaveSettings)
```
Comment (zameelnp) [inline src/ui/src/app/dashboard/dashboard-body/main/leave-management/update-leave-settings/update-leave-settings.component.ts:36]: remove

File: src/ui/src/app/dashboard/dashboard-body/side-nav/side-nav.component.ts [hunk 5]
```diff
@@ -13,5 +13,8 @@ import { EmployeeService } from '../../../service/employee.service';
 export class SideNavComponent {
   userService=inject(WolfDenService)
   employeeService=inject(EmployeeService)
-
+  
+ngOnInit(){
+  console.log('hi',this.userService.role)
```
Comment (zameelnp) [inline src/ui/src/app/dashboard/dashboard-body/side-nav/side-nav.component.ts:18]: ?

File: src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs [hunk 6]
```diff
@@ -10,20 +15,32 @@
...
+        private readonly string _apiKey = configuration["BrevoApi:ApiKey"];
+        private readonly string _senderEmail = configuration["BrevoApi:SenderEmail"];
+        private readonly string _senderName = configuration["BrevoApi:SenderName"];
+        private readonly ManagerEmailFinder _emailFinder = emailFinder;
+        private readonly Email _email = email;
+        private readonly UserManager<User> _userManager = userManager;
         public async Task<bool> Handle(AddLeaveRequestForEmployeeByAdmin request, CancellationToken cancellationToken)
         {
             Employee employee = await _context.Employees.FirstOrDefaultAsync(x => x.EmployeeCode == request.EmployeeCode, cancellationToken);
+            Employee admin = await _context.Employees.FirstOrDefaultAsync(x => x.Id == request.AdminId, cancellationToken);
```
[the remark this answers is not in the document]
Comment (mukhil-geeky) [reply src/services/WolfDen.Application/Requests/Commands/LeaveManagement/AddLeaveRequestForEmployeeByAdmin/AddLeaveRequestForEmployeeByAdminHandler.cs:31]: done

File: src/ui/src/app/dashboard/dashboard-body/side-nav/side-nav.component.ts (hunk 5 above)
Comment (mukhil-geeky) [reply src/ui/src/app/dashboard/dashboard-body/side-nav/side-nav.component.ts:18]: There was an issue in getting userService.role. now it is removed. it was used for checking.

**Review by zameelnp — APPROVED** [review]

**Pull request merged**